client/src/scenes/Battle.ts

```diff
@@ -409,33 +453,18 @@ export default class Battle extends Phaser.Scene {
       .layout()
       .popUp(500);
 
-    this.dialog.on("button.click", function (button, groupName, index) {
-      if (button.name === "fightButton") {
-        // Check if the 'Fight' button was clicked
-        console.log("Fight clicked");
-        // onclick call back
-      }
-
-      if (button.name === "option1") {
-        console.log("Option 1 clicked");
-        // onclick call back
-      }
-
-      if (button.name === "option2") {
-        console.log("Option 2 clicked");
-        // onclick call back
-      }
-
-      if (button.name === "option3") {
-        console.log("Option 3 clicked");
-        // onclick call back
-      }
-
-      if (button.name === "option4") {
-        console.log("Option 4 clicked");
-        // onclick call back
-      }
-    });
+    this.dialog.on(
+      "button.click",
```

Comment:
not sure if this is a property, but if its an event message please update to buttonClick

Reply:
it this also fixed unfortunately